worlds/soe/__init__.py

```diff
@@ -208,14 +208,68 @@ def stage_assert_generate(cls, world):
             raise FileNotFoundError(rom_file)
 
     def create_regions(self):
+        # exclude 'hidden' on easy
+        max_difficulty = 1 if self.world.difficulty[self.player] == Difficulty.option_easy else 256
+
         # TODO: generate *some* regions from locations' requirements?
         r = Region('Menu', RegionType.Generic, 'Menu', self.player, self.world)
         r.exits = [Entrance(self.player, 'New Game', r)]
         self.world.regions += [r]
 
+        # group locations into spheres (1, 2, 3+ at index 0, 1, 2)
+        spheres: typing.Dict[int, typing.Dict[int, typing.List[SoELocation]]] = {}
+        for loc in _locations:
+            spheres.setdefault(min(2, len(loc.requires)), {}).setdefault(loc.type, []).append(
+                SoELocation(self.player, loc.name, self.location_name_to_id[loc.name], r,
+                            loc.difficulty > max_difficulty))
+
+        # location balancing data
+        trash_fills: typing.Dict[int, typing.Dict[int, typing.Tuple[int, int, int, int]]] = {
+            0: {pyevermizer.CHECK_GOURD: (20, 40, 40, 40)},  # remove up to 40 gourds from sphere 1
+            1: {pyevermizer.CHECK_GOURD: (70, 90, 90, 90)},  # remove up to 90 gourds from sphere 2
+        }
+
+        # mark some as excluded based on numbers above
+        for trash_sphere, fills in trash_fills.items():
+            for typ, counts in fills.items():
+                count = counts[self.world.difficulty[self.player].value]
+                for location in self.world.random.sample(spheres[trash_sphere][typ], count):
+                    location.progress_type = LocationProgressType.EXCLUDED
```

Comment:
If you add the name of the location to world.exclude_locations[player].value then both the rule and type will be set after set_rules/before generate_basic for you, and they will appear in the spoiler log.

Reply:
here i absolutely do not want it in the spoiler log